### DISCUSSION TITLE
Anyone figure out a clever way to integrate buttons/tabs/selectors with Rich?

### DISCUSSION THREAD

**Discussion by LPX55** [Textualize/rich#1332] [Q&A]
Absolutely love Rich, love what you guys have made. I was wondering if anyone had a clever 3rd party / hack to allow for more interactivity as we all patiently wait for the TUI :P Only somewhat successful method I've found was through keyboard input, which was surprising considering the number of projects that use Rich -- and even so, it's not the most user-friendly for the non-technical crowd.  

Would an integration with something like py_cui or something similar be possible? Or would it be a daunting task not worth pursuing?

**Comment by willmcgugan** [comment]
Rich doesn't provide any explicit support for TUIs / full screen interfaces, but you can [capture output](https://rich.readthedocs.io/en/latest/console.html#capturing-output). Some people have combined this with the keyboard module to make things more dynamic, so it is possible with a bit of work.

Or you could wait for [Textual](https://github.com/willmcgugan/textual)

**Reply by LPX55** [reply]
Figured you'd say that 😅 Just thought maybe someone in the community had their unique little "hacks" to share. I'm working on one right now, hopefully it'll work as planned and I'll be able to share. Unfortunately I'm working on a time-sensitive project and I found out about prompt-toolkit way too late, ha. Anyways, you are a legend. Rich is truly a gem for the Python community.

P.S. ever consider changing / rebranding the name? Often times its a bit difficult to do a proper search on Google and GitHub because the name is so widely used in all sorts of context. Just wondering ;) From a marketing point, it might do ya some real good!
